examples/react-native/v12/TestApp/src/components/data-types/geospatial.tsx

```diff
@@ -0,0 +1,300 @@
+    if (!companies.length) {
+      // Add geospatial objects to realm.
+      writeNewCompany({_id: 6, location: new MyGeoPoint(-122.35, 47.68)});
+      writeNewCompany({_id: 9, location: new MyGeoPoint(-121.85, 47.9)});
+    }
+  }, []);
+
+  type CompanyProps = {
+    _id: number;
+    location: MyGeoPoint;
+  };
+
+  const writeNewCompany = ({_id, location}: CompanyProps) => {
+    // Add geospatial object to realm.
+    realm.write(() => {
+      realm.create(Company, {
+        _id,
+        location,
+      });
+    });
+  };
+
+  return (
+    <View>
+      <Geocircle />
+      <Geobox />
+      <Geopolygon />
+    </View>
+  );
+}
+// :snippet-end:
+
+// :snippet-start: geocircle
+// :uncomment-start:
+// import React from 'react';
+// import {View, Text} from 'react-native';
+// import {GeoCircle, GeoPoint, kmToRadians} from 'realm';
+// import {useQuery} from '@realm/react';
+// :uncomment-end:
+
+function Geocircle(): JSX.Element {
+  // Define a GeoCircle
+  const smallCircle: GeoCircle = {
+    center: [-121.9, 47.3],
+    // The GeoCircle radius is measured in radians.
+    // This radian distance corresponds with 0.25 degrees.
+    distance: 0.004363323,
+  };
+
+  // Realm provides `kmToRadians` and `miToRadians`
+  // to convert these measurements. Import the relevant
+  // convenience method for your app's needs.
+  const radiusFromKm = kmToRadians(44.4);
+
+  // Define a GeoPoint within a GeoCircle
+  const largeCircleCenter: GeoPoint = {
+    longitude: -122.6,
+    latitude: 47.8,
+  };
+
+  const largeCircle: GeoCircle = {
+    center: largeCircleCenter,
+    distance: radiusFromKm,
+  };
+
+  // Query geospatial data
+  const companiesInSmallCircle = useQuery(Company).filtered(
+    'location geoWithin $0',
+    smallCircle,
+  );
+
+  // Query geospatial data
+  const companiesInLargeCircle = useQuery(Company).filtered(
+    'location geoWithin $0',
+    largeCircle,
+  );
+
+  return (
+    <View>
+      <Text>Small circle: {companiesInSmallCircle.length}</Text>
+      <Text>Large circle: {companiesInLargeCircle.length}</Text>
+    </View>
+  );
+}
+// :snippet-end:
+
+// :snippet-start: geobox
+// :uncomment-start:
+// import React from 'react';
+// import {View, Text} from 'react-native';
+// import {GeoBox, GeoPoint} from 'realm';
+// import {useQuery} from '@realm/react';
+// :uncomment-end:
+
+function Geobox(): JSX.Element {
+  // Define a GeoBox
+  const largeBox: GeoBox = {
+    bottomLeft: [-122.7, 47.3],
+    topRight: [-122.1, 48.1],
+  };
+
+  // Define GeoBox corners
+  const smallBoxBottomLeft: GeoPoint = {
+    longitude: -122.4,
+    latitude: 47.5,
+  };
+  const smallBoxTopRight: GeoPoint = {
+    longitude: -121.8,
+    latitude: 47.9,
+  };
+  const smallBox: GeoBox = {
+    bottomLeft: smallBoxBottomLeft,
+    topRight: smallBoxTopRight,
+  };
+
+  // Query geospatial data
+  const companiesInLargeBox = useQuery(Company).filtered(
+    'location geoWithin $0',
+    largeBox,
+  );
```

Comment:
```suggestion
  const companiesInLargeBox = useQuery(Company, (col) => col.filtered(
    'location geoWithin $0',
    largeBox,
  ), [largeBox]);
```
Lets start making use of this syntax for `useQuery`